src/webgl/framebuffer.js

```diff
@@ -45,90 +35,176 @@ export default class Framebuffer extends Resource {
 
   /* eslint-disable max-statements */
   initialize({
+    attachments = {},
     width = 1,
     height = 1,
-    depth = true,
-    minFilter = GL.NEAREST,
-    magFilter = GL.NEAREST,
+    color = true,
     format = GL.RGBA,
-    type = GL.UNSIGNED_BYTE
+    type = GL.UNSIGNED_BYTE,
+    parameters = {
+      [GL.TEXTURE_MIN_FILTER]: GL.NEAREST,
+      [GL.TEXTURE_MAG_FILTER]: GL.NEAREST
+    },
+    depth = true
   }) {
+    const {gl} = this;
+
     assert(width >= 0 && height >= 0, 'Width and height need to be integers');
     if (width === this._width && height === this._height) {
       return;
     }
 
     log.log(2, `Resizing framebuffer ${this.id} to ${width}x${height}`);
 
-    this.opts.colorBuffer = null;
-    this.opts.depthBuffer = null;
-    this.opts.stencilBuffer = null;
-    this.opts.texture = null;
-
-    // TODO - do we need to reallocate the framebuffer?
-    const colorBuffer = new Texture2D(this.gl, {
-      data: null,
-      format,
-      type,
-      width,
-      height,
-      [GL.TEXTURE_MIN_FILTER]: this.minFilter,
-      [GL.TEXTURE_MAG_FILTER]: this.magFilter
-    });
-
-    this.attachTexture({
-      attachment: GL.COLOR_ATTACHMENT0,
-      texture: colorBuffer
-    });
-
-    if (this.opts.colorBuffer) {
-      // this.opts.colorBuffer.delete();
+    this.colorBuffer = null;
+    this.depthBuffer = null;
+    this.stencilBuffer = null;
+    this.texture = null;
+
+    attachments = Object.assign({}, attachments);
+
+    // Add a color buffer if requested and not supplied
+    if (color && !attachments[gl.COLOR_ATTACHMENT0]) {
+      const colorBuffer = new Texture2D(this.gl, {
+        data: null,
+        format,
+        type,
+        width,
+        height,
+        parameters
+      });
+
+      attachments[gl.COLOR_ATTACHMENT0] = colorBuffer;
+
+      if (this.texture) {
+        // this.texture.delete();
+      }
+      this.texture = colorBuffer;
     }
-    this.opts.colorBuffer = colorBuffer;
-    this.opts.texture = colorBuffer;
 
-    // Add a depth buffer if requested
-    if (depth) {
+    // Add a depth buffer if requested and not supplied
+    if (depth && !attachments[gl.DEPTH_ATTACHMENT]) {
       const depthBuffer = new Renderbuffer(this.gl, {
-        format: GL.DEPTH_COMPONENT16,
+        format: gl.DEPTH_COMPONENT16,
         width,
         height
       });
-      this.attachRenderbuffer({
-        attachment: GL.DEPTH_ATTACHMENT,
-        renderbuffer: depthBuffer
-      });
+      attachments[gl.DEPTH_ATTACHMENT] = depthBuffer;
 
-      if (this.opts.depthBuffer) {
-        // this.opts.depthBuffer.delete();
+      if (this.depthBuffer) {
```

Comment:
> // this.opts.depthBuffer.delete(); 
> Remove this?

The idea is that the autogenerated targets would be autodeleted when e.g. resizing the framebuffer. Just need a little more time to get the logic correct.